HubSpot.NET/Api/Contact/Dto/ContactHubSpotModel.cs

```diff
@@ -17,43 +19,175 @@ public class ContactHubSpotModel : IHubSpotModel
         [DataMember(Name = "vid")]
         [IgnoreDataMember]
         public long? Id { get; set; }
+
+        private string _Email;
         [DataMember(Name = "email")]
-        public string Email { get; set; }
+        public string Email {
+            set {
+                _Email = value;
+                if (Properties.ContainsKey("email"))
```

Comment:
OK I think I follow. Would this mean you'd need to add this method to every property? (including on custom models)?

Reply:
In our testing this seemed to only be an issue with the `Contact` models, but @rezalas was actually the one to create this transform so he could give a more in depth answer.

Reply:
Cool. Would be definitely good to hear more about that, I think it would be a shame if user's had to implement this themselves for all their custom models.

Reply:
The only reason I added these was to support ease of use for the models and reduce how much "inner workings" knowledge a dev required to use contacts. Previously the values were available because the dynamic adapter was retrieving them, but with the changes to not need that adapter the values are all stored in the Properties dictionary (line 188) which is how they're actually returned by HubSpot. We don't have to continue using them if we don't want, but I thought this was a "most used" ease-of-use method for commonly referenced properties. For everything else, devs can retrieve the value of a property by passing the key to the dictionary.

Reply:
I think it would be quite a big regression if devs had to either write these adapters or access a dictionary directly rather than a strongly typed model. Do you think it would be difficult to move to a strongly typed model without the need for the adapters?

Reply:
Do you have an example of a custom model that that extends the contract returned by HubSpot? I'm curious to know what use cases there are for sending or receiving data that aren't a part of the HubSpot contracts.

Reply:
Depending on what you're thinking of with a strongly typed model, it may be possible. The model we have now is strongly typed, but we're also meeting the contract with the hubspot API at the model level for simplicity. If you look at the converter PropertyTransportModel<T>.ToPropertyTransportModel(T model) you'll see that I wrote a dynamic adapter as well that can be used, we just have to add an overload for the HubSpotContactApi to support taking in custom models that can be converted.